homeassistant/components/homematicip_cloud/__init__.py

```diff
@@ -31,15 +33,12 @@
     'climate',
 ]
 
-CONF_NAME = 'name'
-CONF_ACCESSPOINT = 'accesspoint'
-CONF_AUTHTOKEN = 'authtoken'
-
+# All parameter are optional in order to support config entry loading
```

Comment:
Config entries doesn't use the config schema by default, so we shouldn't change this. It doesn't make sense to allow a config without necessary items. Instead check if the config dict for the DOMAIN is empty. If it's empty we know the component isn't being setup via normal bootstrap but from entry.

Reply:
@MartinHjelmare First thank you very much for your time to start this review! 
I know the implementation is far from a clean state.
This was a left over - misunderstanding - will be reverted back!